larrecodnn/NuGraph/Tools/LoaderToolBase.h

```diff
@@ -0,0 +1,70 @@
+#ifndef LOADERTOOLBASE_H
+#define LOADERTOOLBASE_H
+
+// art TOOL
+#include "art/Utilities/ToolMacros.h"
+#include "art/Utilities/make_tool.h"
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "fhiclcpp/ParameterSet.h"
+
+#include "lardataobj/RecoBase/Hit.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+using std::string;
+using std::vector;
+
+struct NuGraphInput {
+  NuGraphInput(string s, vector<int32_t> vi)
+    : input_name(s), isInt(true), input_int32_vec(vi), input_float_vec(vector<float>())
```

Comment:
Suggest moving named vectors to avoid additional copy; and no need to explicitly initialize vectors to an empty vector:

```suggestion
    : input_name(s), isInt(true), input_int32_vec(std::move(vi))
```